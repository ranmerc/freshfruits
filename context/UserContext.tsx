import { createContext, useContext, ReactNode } from "react";
import { useLocalStorage } from "usehooks-ts";

export interface UserContextType {
  username: string;
  setUserName: (user: string) => void;
}

export const UserContext = createContext<UserContextType | undefined>(
  undefined
);

export function useUserContext() {
  const context = useContext(UserContext);

  if (!context) {
    throw new Error("useUserContext must be used within a UserProvider");
  }

  return context;
}

export function UserProvider({ children }: { children: ReactNode }) {
  const [username, setUserName] = useLocalStorage<string>("username", "");

  return (
    <UserContext.Provider value={{ username, setUserName }}>
      {children}
    </UserContext.Provider>
  );
}
