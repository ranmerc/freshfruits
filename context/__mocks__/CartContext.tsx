export const CartContext = jest.fn();

export const useCartContext = jest.fn();

export const CartProvider = jest.fn();
