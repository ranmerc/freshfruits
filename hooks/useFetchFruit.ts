import axios from "axios";
import { useQuery } from "@tanstack/react-query";
import FruitAPIResponse from "@/types/FruitAPIResponse";

const fetcher = async (id: number) => {
  const response = await axios.get<FruitAPIResponse>(`/api/fruit?id=${id}`);
  return response.data;
};

export default function useFetchFruit(id: number) {
  return useQuery<FruitAPIResponse>({
    queryKey: ["fruit", id],
    queryFn: () => fetcher(id),
  });
}
