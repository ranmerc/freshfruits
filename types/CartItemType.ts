import { Pack } from "./PriceData";

export default interface CartItemType {
  fruitId: number;
  selectedPackId: number;
  quantity: number;
  pack: Pack;
}
