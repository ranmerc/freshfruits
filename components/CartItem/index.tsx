import React from "react";
import {
  Card,
  CardContent,
  CardMedia,
  Typography,
  Stack,
  Alert,
} from "@mui/material";
import CartItem from "@/types/CartItemType";
import CartItemType from "@/types/CartItemType";
import { useCartContext } from "@/context/CartContext";
import useFetchFruit from "@/hooks/useFetchFruit";
import Link from "next/link";
import calculateDiscountPercentage from "@/utils/calculateDiscountPercentage";
import Counter from "../Counter";

export default function CartItem({ item }: { item: CartItemType }) {
  const { data, isError, isSuccess, isLoading } = useFetchFruit(item.fruitId);

  const { addItemToCart, removeItemFromCart } = useCartContext();

  const discount = calculateDiscountPercentage(
    item.pack.discountPrice,
    item.pack.price
  );

  const handleIncreaseQuantity = () => {
    addItemToCart({
      fruitId: item.fruitId,
      pack: item.pack,
      quantity: 1,
      selectedPackId: item.selectedPackId,
    });
  };

  const handleDecreaseQuantity = () => {
    removeItemFromCart({
      fruitId: item.fruitId,
      selectedPackId: item.selectedPackId,
    });
  };

  if (isLoading) {
    <Alert severity="info">Loading item...</Alert>;
  }

  if (isError) {
    <Alert severity="error">Error loading item!</Alert>;
  }

  if (isSuccess && data && data.data) {
    const { name, images } = data.data;
    return (
      <Card
        sx={{
          display: "flex",
          p: 3,
          columnGap: 2,
          flexWrap: "wrap",
          flexBasis: 1,
        }}
      >
        <CardMedia
          sx={{
            width: { xs: "100%", md: 150 },
            height: 150,
          }}
          image={images[0]}
        />
        <CardContent
          sx={{
            display: "flex",
            p: 0,
            flexGrow: 1,
            justifyContent: "space-between",
            flexWrap: "wrap",
          }}
        >
          <Stack rowGap={0.5}>
            <Typography
              variant="h5"
              fontWeight={"bold"}
              textTransform={"capitalize"}
            >
              <Link
                href={`/fruit/${item.fruitId}`}
                style={{ textDecoration: "none" }}
                color="inherit"
              >
                {`${name} (${item.pack.quantity}
              ${item.pack.type === "count" ? "Nos" : "g"})`}
              </Link>
            </Typography>
            {!!discount && (
              <Typography
                variant="caption"
                sx={{
                  bgcolor: "secondary.main",
                  color: "secondary.contrastText",
                  width: "fit-content",
                  px: 0.5,
                }}
              >{`${discount}% off`}</Typography>
            )}
            <Stack direction={"row"} alignItems={"baseline"} columnGap={0.5}>
              <Typography
                variant="body1"
                component={"span"}
                fontSize={"1.2rem"}
              >
                ₹{item.pack.discountPrice}
              </Typography>
              {item.pack.discountPrice !== item.pack.price && (
                <Typography
                  variant="body2"
                  component={"span"}
                  sx={{ textDecoration: "line-through" }}
                >
                  ₹{item.pack.price}
                </Typography>
              )}
            </Stack>
            <Stack
              direction={"row"}
              alignItems={"center"}
              sx={{
                bgcolor: "secondary.main",
                width: "fit-content",
                borderRadius: 2,
              }}
            >
              <Counter
                count={item.quantity}
                decrementLabel="Remove one item"
                incrementLabel="Add one item"
                onDecrement={handleDecreaseQuantity}
                onIncrement={handleIncreaseQuantity}
                incrementDisabled={!item.pack.inStock}
              />
            </Stack>
          </Stack>

          <Typography
            variant="body1"
            component="span"
            fontSize={"1.2rem"}
            fontWeight={"bold"}
          >
            ₹{item.pack.discountPrice * item.quantity}
          </Typography>
        </CardContent>
      </Card>
    );
  }
}
