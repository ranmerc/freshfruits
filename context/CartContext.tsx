import { useLocalStorage } from "usehooks-ts";
import {
  createContext,
  useContext,
  ReactNode,
  useReducer,
  Dispatch,
  useCallback,
} from "react";
import CartItem from "@/types/CartItem";

type ActionType =
  | {
      type: "ADD_TO_CART";
      payload: CartItem;
    }
  | {
      type: "REMOVE_FROM_CART";
      payload: {
        fruitId: number;
        selectedPackId: number;
      };
    }
  | {
      type: "CLEAR_CART";
    };

interface CartContextType {
  cartItems: CartItem[];
  cartDispatch: Dispatch<ActionType>;
}

const CartContext = createContext<CartContextType | undefined>(undefined);

export function useCartContext() {
  const context = useContext(CartContext);

  if (!context) {
    throw new Error("useCartContext must be used within a CartProvider");
  }

  return context;
}

function cartReducer(state: CartItem[], action: ActionType) {
  switch (action.type) {
    case "ADD_TO_CART": {
      const { fruitId, selectedPackId, quantity } = action.payload;

      const existingItemIndex = state.findIndex(
        (item) =>
          item.fruitId === fruitId && item.selectedPackId === selectedPackId
      );

      if (existingItemIndex !== -1) {
        const newState = [...state];
        newState[existingItemIndex].quantity += quantity;
        return newState;
      }

      return [...state, action.payload];
    }

    case "REMOVE_FROM_CART": {
      const { fruitId, selectedPackId } = action.payload;

      const existingItemIndex = state.findIndex(
        (item) =>
          item.fruitId === fruitId && item.selectedPackId === selectedPackId
      );

      if (existingItemIndex === -1) {
        return state;
      }

      const newState = [...state];
      newState[existingItemIndex].quantity -= 1;

      if (newState[existingItemIndex].quantity === 0) {
        newState.splice(existingItemIndex, 1);
      }

      return newState;
    }

    case "CLEAR_CART":
      return [];
    default:
      return state;
  }
}

export function CartProvider({ children }: { children: ReactNode }) {
  const [savedCartItems, setSavedCartItems] = useLocalStorage<CartItem[]>(
    "cart-items",
    []
  );

  const reducerLocalStorage = useCallback(
    (state: CartItem[], action: ActionType) => {
      const newState = cartReducer(state, action);

      setSavedCartItems(newState);

      return newState;
    },
    [setSavedCartItems]
  );

  const [state, dispatch] = useReducer(reducerLocalStorage, savedCartItems);

  return (
    <CartContext.Provider value={{ cartItems: state, cartDispatch: dispatch }}>
      {children}
    </CartContext.Provider>
  );
}
