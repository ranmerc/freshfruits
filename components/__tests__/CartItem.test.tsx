import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CartItem from "../CartItem";
import useFetchFruit from "@/hooks/useFetchFruit";
import { ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { useCartContext } from "@/context/CartContext";
import CartItemType from "@/types/CartItemType";
import userEvent from "@testing-library/user-event";

jest.mock("@/hooks/useFetchFruit");
jest.mock("@/context/CartContext");

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const mockedUseFetchFruit = useFetchFruit as jest.Mock;
const mockedUseCartContext = useCartContext as jest.Mock;

const wrapper = ({ children }: { children: ReactNode }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const cartItemValue = (details: Partial<CartItemType>): CartItemType => {
  return {
    fruitId: details.fruitId || 1,
    pack: {
      discountPrice: details.pack ? details.pack.discountPrice : 30,
      inStock: details.pack ? details.pack.inStock : false,
      price: details.pack ? details.pack.price : 35,
      quantity: details.pack ? details.pack.quantity : 2,
      type: details.pack ? details.pack.type : "count",
    },
    quantity: details.quantity || 1,
    selectedPackId: details.selectedPackId || 1,
  };
};

describe("CartItem component", () => {
  it("Renders loading state", () => {
    mockedUseFetchFruit.mockImplementation((id: number) => ({
      isLoading: true,
      isError: false,
      data: null,
      isSuccess: false,
    }));

    mockedUseCartContext.mockReturnValue({
      addItemToCart: jest.fn(),
      removeItemFromCart: jest.fn(),
    });

    render(<CartItem item={cartItemValue({})} />, {
      wrapper: wrapper,
    });

    expect(screen.getByText("Loading item...")).toBeInTheDocument();
  });

  it("Renders error state", () => {
    mockedUseFetchFruit.mockImplementation((id: number) => ({
      isLoading: false,
      isError: true,
      data: null,
      isSuccess: false,
    }));

    const addFunction = jest.fn();
    const removeFunction = jest.fn();
    mockedUseCartContext.mockReturnValue({
      addItemToCart: addFunction,
      removeItemFromCart: removeFunction,
    });

    render(<CartItem item={cartItemValue({})} />, {
      wrapper: wrapper,
    });

    expect(screen.getByText("Error loading item!")).toBeInTheDocument();
  });

  it("Renders correct fruit details - 1", () => {
    mockedUseFetchFruit.mockImplementation(() => ({
      isLoading: false,
      isError: false,
      data: {
        data: {
          name: "Fruit",
          images: ["image.jpeg"],
        },
      },
      isSuccess: true,
    }));

    mockedUseCartContext.mockReturnValue({
      addItemToCart: jest.fn(),
      removeItemFromCart: jest.fn(),
    });

    render(
      <CartItem
        item={cartItemValue({
          fruitId: 3,
          pack: {
            quantity: 5,
            inStock: true,
            price: 10,
            discountPrice: 5,
            type: "count",
          },
        })}
      />,
      {
        wrapper: wrapper,
      }
    );

    // image
    expect(
      screen.getByRole("img", {
        name: "Fruit",
      })
    ).toBeInTheDocument();

    // fruit heading
    expect(
      screen.getByRole("heading", {
        name: "Fruit (5 Nos)",
      })
    );

    // link to fruit
    expect(
      screen.getByRole("link", {
        name: "Fruit (5 Nos)",
      })
    ).toHaveAttribute("href", "/fruit/3");

    // discount and subtotal price
    expect(screen.getByText("50% off")).toBeInTheDocument();

    // discount price
    expect(screen.getAllByText(/₹5/i)).toHaveLength(2);

    // actual price
    expect(screen.getByText(/₹10/i)).toHaveStyle(
      "text-decoration: line-through"
    );
  });

  it("Renders correct fruit details - 2", () => {
    mockedUseFetchFruit.mockImplementation(() => ({
      isLoading: false,
      isError: false,
      data: {
        data: {
          name: "Apple",
          images: ["img.jpeg"],
        },
      },
      isSuccess: true,
    }));

    mockedUseCartContext.mockReturnValue({
      addItemToCart: jest.fn(),
      removeItemFromCart: jest.fn(),
    });

    render(
      <CartItem
        item={cartItemValue({
          fruitId: 2,
          pack: {
            quantity: 500,
            inStock: true,
            price: 220,
            discountPrice: 200,
            type: "weight",
          },
          quantity: 4,
        })}
      />,
      {
        wrapper: wrapper,
      }
    );

    // image
    expect(
      screen.getByRole("img", {
        name: "Apple",
      })
    ).toBeInTheDocument();

    // fruit heading
    expect(
      screen.getByRole("heading", {
        name: "Apple (500 g)",
      })
    );

    // link to fruit
    expect(
      screen.getByRole("link", {
        name: "Apple (500 g)",
      })
    ).toHaveAttribute("href", "/fruit/2");

    // discount
    expect(screen.getByText("9% off")).toBeInTheDocument();

    // discount price
    expect(screen.getByText(/₹200/i)).toBeInTheDocument();

    // subtotal price
    expect(screen.getByText(/₹800/i)).toBeInTheDocument();

    // actual price
    expect(screen.getByText(/₹220/i)).toHaveStyle(
      "text-decoration: line-through"
    );
  });

  it("Functions correctly", async () => {
    mockedUseFetchFruit.mockImplementation(() => ({
      isLoading: false,
      isError: false,
      data: {
        data: {
          name: "Apple",
          images: ["img.jpeg"],
        },
      },
      isSuccess: true,
    }));

    const addItemToCart = jest.fn();
    const removeItemFromCart = jest.fn();
    mockedUseCartContext.mockReturnValue({
      addItemToCart: addItemToCart,
      removeItemFromCart: removeItemFromCart,
    });

    render(
      <CartItem
        item={cartItemValue({
          fruitId: 2,
          pack: {
            quantity: 500,
            inStock: true,
            price: 220,
            discountPrice: 200,
            type: "weight",
          },
          quantity: 4,
        })}
      />,
      {
        wrapper: wrapper,
      }
    );

    const incrementButton = screen.getByRole("button", {
      name: "Add one item",
    });

    const decrementButton = screen.getByRole("button", {
      name: "Remove one item",
    });

    await userEvent.click(incrementButton);
    await userEvent.click(decrementButton);

    expect(addItemToCart).toHaveBeenCalledTimes(1);
    expect(removeItemFromCart).toHaveBeenCalledTimes(1);
  });
});
