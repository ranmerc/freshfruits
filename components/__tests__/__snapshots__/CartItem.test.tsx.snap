// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component Renders correct fruit details - 1 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-lk0ebk-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardMedia-root css-18e91k1-MuiCardMedia-root"
    role="img"
    style={
      {
        "backgroundImage": "url("image.jpeg")",
      }
    }
    title="Fruit"
  />
  <div
    className="MuiCardContent-root css-14t07bv-MuiCardContent-root"
  >
    <div
      className="MuiStack-root css-1tc234f-MuiStack-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-1hkt7f5-MuiTypography-root"
      >
        <a
          color="inherit"
          href="/fruit/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "textDecoration": "none",
            }
          }
        >
          Fruit (2
              Nos)
        </a>
      </h5>
      <span
        className="MuiTypography-root MuiTypography-caption css-qss0wv-MuiTypography-root"
      >
        14% off
      </span>
      <div
        className="MuiStack-root css-1f7x521-MuiStack-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 css-b02fjb-MuiTypography-root"
        >
          ₹
          30
        </span>
        <span
          className="MuiTypography-root MuiTypography-body2 css-qc7gyi-MuiTypography-root"
        >
          ₹
          35
        </span>
      </div>
      <div
        className="MuiStack-root css-w62c0k-MuiStack-root"
      >
        <button
          aria-label="Remove one item"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RemoveCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
            />
          </svg>
        </button>
        <p
          className="MuiTypography-root MuiTypography-body1 css-b02fjb-MuiTypography-root"
        >
          1
        </p>
        <button
          aria-label="Add one item"
          className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
        </button>
      </div>
    </div>
    <span
      className="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
    >
      ₹
      30
    </span>
  </div>
</div>
`;

exports[`CartItem component Renders correct fruit details - 2 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-lk0ebk-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardMedia-root css-18e91k1-MuiCardMedia-root"
    role="img"
    style={
      {
        "backgroundImage": "url("img.jpeg")",
      }
    }
    title="Apple"
  />
  <div
    className="MuiCardContent-root css-14t07bv-MuiCardContent-root"
  >
    <div
      className="MuiStack-root css-1tc234f-MuiStack-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-1hkt7f5-MuiTypography-root"
      >
        <a
          color="inherit"
          href="/fruit/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "textDecoration": "none",
            }
          }
        >
          Apple (2
              Nos)
        </a>
      </h5>
      <span
        className="MuiTypography-root MuiTypography-caption css-qss0wv-MuiTypography-root"
      >
        14% off
      </span>
      <div
        className="MuiStack-root css-1f7x521-MuiStack-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 css-b02fjb-MuiTypography-root"
        >
          ₹
          30
        </span>
        <span
          className="MuiTypography-root MuiTypography-body2 css-qc7gyi-MuiTypography-root"
        >
          ₹
          35
        </span>
      </div>
      <div
        className="MuiStack-root css-w62c0k-MuiStack-root"
      >
        <button
          aria-label="Remove one item"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RemoveCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          className="MuiTypography-root MuiTypography-body1 css-b02fjb-MuiTypography-root"
        >
          1
        </p>
        <button
          aria-label="Add one item"
          className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
        </button>
      </div>
    </div>
    <span
      className="MuiTypography-root MuiTypography-body1 css-b6p50a-MuiTypography-root"
    >
      ₹
      30
    </span>
  </div>
</div>
`;

exports[`CartItem component Renders error state 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1hdyuqk-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ErrorOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    Error loading item!
  </div>
</div>
`;

exports[`CartItem component Renders loading state 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="InfoOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    Loading item...
  </div>
</div>
`;
